sonic-net/sonic-utilities
# Static Lag support changes in sonic-utilities

#### What I did
Static lag support for 202211 branch using the following md
https://github.com/sonic-net/SONiC/pull/1039

other dependent pull request

    https://github.com/sonic-net/sonic-buildimage/pull/12360
    https://github.com/sonic-net/sonic-swss/pull/2486
    https://github.com/sonic-net/sonic-utilities/pull/2436

#### How I did it
    1. Add static lag command support in click
    2. update mock_tables so pyrunner execution are success during
compilation

#### How to verify it
    Test cases :-
            Test cases
    1       Create static port channel with static flag     pass
    2       verify static has option flag true or false     pass
    3       Add static member see the portchannel is up     pass
    4       verify teamd is created with loadbalance option by default
    pass
    5       Remove last portchannel member check port channel down  pass
    6       Remove portchannel member check port channel still up   pass
    7       verify teamdctl config dump     pass
    8       verify teamdctl state dump      pass
    9       shutdown the portchannel check the kernel state pass
    10      no shutdown the portchannel check the kernel state      pass
    11      "Check the show output matches the review comment
    root@sonic:~# show inter port
    Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not
    available,
           S - selected, D - deselected, * - not synced
      No.  Team Dev      Protocol     Ports
    -----  ------------  -----------  ------------
        1  PortChannel1  NONE(A)(Up)  Ethernet0(S)
        2  PortChannel2  NONE(A)(Up)  Ethernet8(S)
        4  PortChannel4  NONE(A)(Dw)
    "       pass
    12      teamnl is set to loadbalance    pass
    13      save and reload and verify portchannel is up    pass
    14      "docker restart teamd
    teamd stopped
    swss stopped
    syncd stopped

    swss started
    syncd started
    teamd started"  pass

    15. verify teamd settles doesnt hog cpu with 100% cpu usage pass
  
#### Previous command output (if the output of a command-line utility has changed)
No change to previous output

#### New command output (if the output of a command-line utility has changed)
root@sonic:~# show inter port
Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
       S - selected, D - deselected, * - not synced
  No.  Team Dev      Protocol     Ports
-----  ------------  -----------  ------------
    1  PortChannel1  LACP(A)(Dw)
    2  PortChannel2  NONE(A)(Up)  Ethernet8(S)
